src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    private static final int FIGURE_NAME = 100;
+    private static final int FIGURE = 5;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        if (random.nextInt(FIGURE) == 0) {
```

Comment:
Use a single random number, because this way you could get into several blocks of code if you didn't return.

Reply:
And change for switch, like:
int number = random.nextInt(FIGURE_COUNT);
switch (number) {
    case 0: 
         return getRandomCircle();
    case 1:
         ....
}